### ISSUE TITLE
[icu] icu/*: doesn't support arch=x86_64h

### ISSUE THREAD

**Issue by hoyhoy** [conan-io/conan-center-index#23634]
### Description

Recipe is incorrect.  Doesn't include x86_64h (haswell) support.

### Package and Environment Details

* Package Name/Version: **icu/***
* Operating System+version: **MacOS 14**
* Compiler+version: **LLVM/All**
* Conan version: **conan 2.2.2**
* Python version: **Python 3.7.4**

### Conan profile

[settings]
os=Macos
os_build=Macos
arch=x86_64h
arch_build=x86_64h
compiler=apple-clang
compiler.version=18
compiler.libcxx=libc++
build_type=Release
[options]
[conf]
[build_requires]
[env]

### Steps to reproduce

 conan create --build=missing --version=73.2 .

### Logs

<details><summary>Click to expand log</summary>

```
Requested 32-bit binaries, but none available
```

</details>

**Comment by datalogics-robb** [comment]
I am not familiar with `x86_64h` architecture - can you provide a reference?

**Comment by hoyhoy** [comment]
It's wrong in a lot of places in conan-center recipes.  Doesn't even exist in settings.yml

https://github.com/llvm-mirror/compiler-rt/blob/master/cmake/builtin-config-ix.cmake

**Comment by hoyhoy** [comment]
b2, boost, icu, and settings.yml are all incorrect.

**Comment by valgur** [comment]
@memsharded This sounds like a broader settings.yaml issue. I suggest moving it to conan-io/conan. Any CCI issues can be addressed after that.

**Comment by hoyhoy** [comment]
Yeah, this is what llvm and Xcode are doing to specify post-2013 x86 chips on MacOS now.
